### PR TITLE
xml and json changes

### PR DESCRIPTION
**Pull request by BrentHoltsclaw**
Modified XML and JSON logging to use a common class.  Cleaned up some code with chipsec_main.

### CODE REVIEW

File: chipsec/testcase.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            return FAIL
+        elif len(summary['warnings']) != 0:
+            return WARNING
+        elif len(summary['skipped']) != 0:
+            return SKIPPED
+        elif len(summary['not applicable']) != 0:
+            return NOTAPPLICABLE
+        elif len(summary['information']) != 0:
+            return INFORMATION
+        else:
```
Comment (frinzell) [inline chipsec/testcase.py]: get_return_code() returns appear to be missing proper scope

**Comment by ErikBjorge** [comment]
Are these XML errors or changes to the format?  Just need the report to be a bit more specific so we can fix the problems.

**Comment by frinzell** [comment]
XML errors loading within a browser.  Appears to be invalid attributes "`failed to run`", "`not applicable`", and "`not implemented`"

**XMLParsing Error: not well-formed**
```
<testsuites><testsuite exceptions="1" failed to run="1" information="0" ...
---------------------------------------------^
```

File: chipsec/logger.py [hunk 2]
```diff
@@ -37,7 +37,8 @@
 import os
 from time import localtime, strftime
 
-from chipsec.xmlout import xmlAux
+from chipsec.testcase import TestCase, ChipsecResults
+#from chipsec.xmlout import xmlAux
```
Comment (ErikBjorge) [inline chipsec/logger.py]: Remove dead code for xmlAux instead of leaving commented.

File: chipsec/logger.py [hunk 3]
```diff
@@ -258,13 +254,14 @@ def log( self, text):
 
     def _log(self, text, color, isStatus):
         """Internal method for logging"""
+        if self.Results.get_current() != None:
```
Comment (ErikBjorge) [inline chipsec/logger.py]: Use 'is not' instead of != when checking against None.

File: chipsec/logger.py [hunk 4]
```diff
@@ -392,20 +380,21 @@ def start_test( self, test_name ):
         text = text + "[x][ Module: " + test_name + "\n"
         text = text + "[x][ ======================================================================="
         self._log(text, BLUE, True)
-        self.xmlAux.start_test( test_name )
-
+        if self.Results.get_current() != None:
```
Comment (ErikBjorge) [inline chipsec/logger.py]: See above comment about None

File: chipsec/logger.py [hunk 5]
```diff
@@ -392,20 +380,21 @@ def start_test( self, test_name ):
...
 
     def start_module( self, module_name ):
         """Displays a banner for the module name provided."""
         #text = "\n[*] start module: %s" % module_name
         #self._log(text, WHITE, None)
         self.log( "\n[*] running module: %s" % module_name )
-        self.xmlAux.start_module( module_name )
 
     def end_module( self, module_name ):
+        if self.Results.get_current() != None:
```
Comment (ErikBjorge) [inline chipsec/logger.py]: Same None comment

File: chipsec_main.py [hunk 6]
```diff
@@ -346,71 +265,43 @@ def print_loaded_modules(self):
...
+            logger().end_module( modx.get_name() )
 
         if self._json_out:
-            results_json = json.dumps(results, sort_keys=True, indent=2, separators=(',', ': '))
-            chipsec.file.write_file(self._json_out, results_json)
+            chipsec.file.write_file(self._json_out, results.json_summary())
+            
+        if self._xml_out:
+            chipsec.file.write_file(self._xml_out, results.xml_full(self._xml_out))
+            #chipsec.file.write_file(self._xml_out, results.xml_summary())
```
Comment (ErikBjorge) [inline chipsec_main.py]: Dead code.

**Comment by frinzell** [comment]
With last changes: **chipsec_main.py** lines 327, 332, 334, 337, 339, 341, 343, 345, 347, and 348 should be...
> logger().

**Comment by BrentHoltsclaw** [comment]
Closing pull request.  Combined as part of PR 454

**Pull request closed without merging**